src/components/navigation-header/components/dropdown-menu/index.tsx

```diff
@@ -0,0 +1,190 @@
+
+  /**
+   * Generates a unique ID for a group of items based on the main menu ID and
+   * the index of the group.
+   */
+  const getItemGroupId = (groupIndex: number): string => {
+    return `${menuId}-itemGroup-${groupIndex}`
+  }
+
+  /**
+   * Handles click interaction with the activator button. When clicked, if the
+   * menu is:
+   *  - open, then it will be closed
+   *  - closed, then it will be opened
+   */
+  const handleClick = () => {
+    setIsOpen(!isOpen)
+  }
+
+  /**
+   * Handles the behavior that should happen when a key is pressed down.
+   * Currently used by both the activator button and each menu item anchor
+   * element. Currently only handles what happens when the Escape is pressed
+   * because all other keyboard interaction is handled by default interactions
+   * with these elements.
+   *
+   * On Escape:
+   *  - the menu is closed, if it is open
+   *  - the activator button is given focus
+   */
+  const handleKeyDown = (e: KeyboardEvent) => {
+    const { isEscapeKey } = deriveKeyEventState(e)
+    if (isEscapeKey) {
+      setIsOpen(false)
+      activatorButtonRef.current.focus()
+    }
+  }
+
+  /**
+   * Handles the start of a mouse hover interaction with the activator button.
+   * When the mouse pointer hovers over the activator button, the menu will be
+   * opened if it is not already open.
+   */
+  const handleMouseEnter = () => {
+    if (!isOpen) {
+      setIsOpen(true)
+    }
+  }
+
+  /**
+   * Handles the end of a mouse hover interaction with the entire menu. If the
+   * menu is open, and the mouse moves outside the bounds either the activator
+   * button or the dropdown menu list, then the menu will be closed.
+   */
+  const handleMouseLeave = () => {
+    if (isOpen) {
+      setIsOpen(false)
+    }
+  }
+
+  return (
+    <div className={s.root} onMouseLeave={handleMouseLeave} ref={menuRef}>
+      <div className={s.activatorWrapper}>
+        <button
+          aria-controls={menuId}
+          aria-expanded={isOpen}
+          className={s.activator}
+          onClick={handleClick}
+          onKeyDown={handleKeyDown}
+          onMouseEnter={handleMouseEnter}
+          ref={activatorButtonRef}
+        >
+          <Text
+            asElement="span"
+            className={s.activatorText}
+            size={200}
+            weight="medium"
+          >
+            {label}
+          </Text>
+          <IconChevronDown16 className={s.activatorIcon} />
+        </button>
+      </div>
+      <div
+        className={s.dropdownContainer}
+        id={menuId}
+        style={{ display: isOpen ? 'block' : 'none' }}
+      >
+        {itemGroups.map((items: NavigationHeaderItem[], groupIndex: number) => {
+          const groupId = getItemGroupId(groupIndex)
+          const isLastItemGroup = groupIndex === numberOfItemGroups - 1
+          const showDivider = numberOfItemGroups > 1 && !isLastItemGroup
+          return (
+            <>
+              <ul className={s.itemGroup}>
```

Comment:
More of just a question, but is it misleading to keyboard users to open the menu when there are groups and only get announced that there is 1 item? Should we flatten this to be a single ul, and use `<li role="separator" />` to render the hr? Or is this a use case to use role group on the unordered list?

In the video following, you can see when I enter the first menu it reads "Developer home list 1 item" but there is actually 3 items within the menu.


https://user-images.githubusercontent.com/825855/161989956-c734c189-80b1-4609-848e-becd2301c6ac.mp4

Reply:
I don't think it's misleading, because we're expressing to sighted users that these are two separate groups of links. When there are more products, sighted users will be able to infer that the top group of links are "general" and the second group of links are "products". Since non-sighted users aren't able to infer the same thing, I think adding a brief label/name/description for each group would definitely be more informative!